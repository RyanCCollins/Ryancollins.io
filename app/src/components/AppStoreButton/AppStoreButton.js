import React, { PropTypes } from 'react';
import './AppStoreButton.scss';
import AppStore from './app-store.png';
import { Thumbnail } from 'react-foundation';

const AppStoreButton = ({
  url
}) => (
  <a className="app-store-button" href={url}>
    <Thumbnail
      className="app-store-button-image"
      src={AppStore}
      alt="App Store Button"
    />
  </a>
);

AppStoreButton.propTypes = {
  url: PropTypes.string.isRequired
};

export default AppStoreButton;
