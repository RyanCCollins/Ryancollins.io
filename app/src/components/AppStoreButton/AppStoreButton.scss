.app-store-button-image {
  max-height: 100px;
  max-width: 180px;
  margin: 20px;
  padding: 10px;
}
