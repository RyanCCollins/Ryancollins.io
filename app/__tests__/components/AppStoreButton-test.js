import React from 'react';
import renderer from 'react/lib/ReactTestRenderer';

import AppStoreButton from '../../src/components/AppStoreButton/AppStoreButton';

// needed because of this:
// https://github.com/facebook/jest/issues/1353
jest.mock('react-dom');

describe('<AppStoreButton />', () => {
  it('renders with required props', () => {
    const component = renderer.create(
      <AppStoreButton url="https://www.ryancollins.io" />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
