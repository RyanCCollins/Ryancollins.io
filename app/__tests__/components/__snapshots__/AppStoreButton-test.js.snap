exports[`<AppStoreButton /> renders with required props 1`] = `
<a
  className="app-store-button"
  href="https://www.ryancollins.io">
  <img
    alt="App Store Button"
    className="thumbnail app-store-button-image"
    src="app/src/components/AppStoreButton/app-store.5fd010aba3dbc491333be65fbaeefc2b.png" />
</a>
`;
